ginivision/src/main/java/net/gini/android/vision/internal/fileimport/AbstractImportImageUrisAsyncTask.java

```diff
@@ -0,0 +1,214 @@
+                        final byte[] bytesFromUri = UriHelper.getBytesFromUri(uri, mContext);
+                        document.setData(bytesFromUri);
+                    } catch (final IOException e) {
+                        LOG.error("Failed to read uri into memory {}", uri);
+                        if (shouldHaltOnError(multiPageDocument,
+                                new ImportedFileValidationException(
+                                        "Failed to read file into memory"))) {
+                            LOG.debug("Halt on error for uri {}", uri);
+                            return null;
+                        }
+                        continue;
+                    }
+                    if (isCancelled()) {
+                        LOG.debug("Import cancelled for uri {}", uri);
+                        return null;
+                    }
+                    // Create Photo
+                    LOG.debug("Create Photo from uri {}", uri);
+                    final Photo photo = PhotoFactory.newPhotoFromDocument(document);
+                    if (isCancelled()) {
+                        LOG.debug("Import cancelled for uri {}", uri);
+                        return null;
+                    }
+                    // Compress Photo
+                    LOG.debug("Compress Photo created from uri {}", uri);
+                    photo.edit().compressByDefault().apply();
+                    if (isCancelled()) {
+                        LOG.debug("Import cancelled for uri {}", uri);
+                        return null;
+                    }
+                    // Save to local storage
+                    LOG.debug("Save compressed Photo to local storage created from uri {}", uri);
+                    final Uri localUri = mGiniVision.internal().getImageDiskStore()
+                            .save(mContext, photo.getData());
+                    if (localUri == null) {
+                        LOG.error("Failed to copy to app storage uri {}", uri);
+                        if (shouldHaltOnError(multiPageDocument,
+                                new ImportedFileValidationException(
+                                        "Failed to copy to app storage"))) {
+                            LOG.debug("Halt on error for uri {}", uri);
+                            return null;
+                        }
+                        continue;
+                    }
+                    if (isCancelled()) {
+                        LOG.debug("Import cancelled for uri {}", uri);
+                        return null;
+                    }
+                    // Create compressed Document
+                    final ImageDocument compressedDocument =
+                            DocumentFactory.newImageDocumentFromPhoto(photo, localUri);
+                    LOG.debug("Compressed ImageDocument created from uri {}", uri);
+                    multiPageDocument.addDocument(compressedDocument);
```

Comment:
Since the point of this `if` is to add a document to the multipage document at the end, you could extract this huge clause to a method and then either return `null` if the document is `null` or add it to the multipage document.